docs/docs/cloud.md

```diff
@@ -5,36 +5,159 @@ title: "Overview"
+  -derivedDataPath ./build
+```
+
+Be sure to note the relative paths of applications, as they will be required for running the tests. In the context of our example, involving the `debug` build, these files can be located at the following paths:
+
+- Application: `/home/john/sample/build/Build/Products/Debug-iphonesimulator/sample.app`
+- Test APK: `/home/john/sample/build/Build/Products/Debug-iphonesimulator/sampleUITests-Runner.app`
+
+One important thing to note is that `*.app` files are actually folders in disguise. In order to transfer them, it's necessary to convert these bundles into standard zip archives:
+
+```shell
+# file structure
+# |
+# |--home
+#    |--john
+#       |--sample <== you are here
+#          |--build  <== derivedData folder
+#          |--sample <== it's your application
+#          ...
+#          |--sample.xcodeproj
+cd build/Build/Products/Debug-iphonesimulator
+zip -r sample.zip sample.app
+zip -r sampleUITests-Runner.zip sampleUITests-runner.app 
+```
+
+Further, we will use these files:
+
+- Application: `/home/john/sample/build/Build/Products/Debug-iphonesimulator/sample.zip`
+- Test APK: `/home/john/sample/build/Build/Products/Debug-iphonesimulator/sampleUITests-Runner.zip`
 
 ### Android
 
-Marathon Cloud supports tests written with UIAutomator, Cucumber, Espresso, and Kaspresso frameworks.
-You will need APK files for both the application and the tests.
+Marathon Cloud supports tests written with **UIAutomator, Cucumber, Espresso, and [Kaspresso](https://github.com/KasperskyLab/Kaspresso) frameworks**.
+
+Before initiating the testing process for your application, you’ll require two APK files: one for the application that’s being tested, and another for the tests themselves. Typically, `debug` variants are utilized for this purpose.
+
+If the primary application resides under the `app/` subproject, you can execute the following command to build both the app and test APK:
+
+```shell
+# file structure
+# |
+# |--home
+#    |--john
+#       |--project <== you are here
+#          |--app  <== it's your primary application
+#          ...
+#          |--build.gragle
+#          |--settings.gradle  
+./gradlew :app:assembleDebug :app:assembleDebugAndroidTest
+```
+
+Be sure to note the relative paths of the test APK and the app APK, as they will be required for running the tests. In the context of our example, involving the `app` project and the `debug` build, these files can be located at the following paths:
+
+- App APK: `/home/john/project/app/build/outputs/apk/debug/app-debug.apk`
+- Test APK: `/home/john/project/app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk`
+
+
+## Samples (optional)
+
+To showcase the advantages of Marathon Cloud compared to other solutions, we've prepared a sample app with 300 tests, out of which 15% are flaky. During the initial run, our platform will gather information about the tests. During the second run, it will optimize it to fit within 15 minutes.
```

Comment:
```suggestion
To showcase the advantages of Marathon Cloud compared to other solutions, we've prepared a sample app with 300 tests, of which 15% are flaky. During the initial run, our platform will gather information about the tests. During the second run, it will optimize it to fit within 15 minutes.
```